data.table setup error at CI macOS job in master

```
  There is a binary version available but the source version is later:
           binary source needs_compilation
data.table 1.13.6 1.14.0              TRUE


...


* installing *source* package ‘data.table’ ...
** package ‘data.table’ successfully unpacked and MD5 sums checked
** using staged installation
zlib 1.2.11 is available ok
R CMD SHLIB supports OpenMP without any extra hint
** libs
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG   -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -fopenmp  -fPIC  -Wall -g -O2  -c assign.c -o assign.o
clang: error: unsupported option '-fopenmp'
make: *** [assign.o] Error 1
ERROR: compilation failed for package ‘data.table’
* removing ‘/Users/runner/Rlib/data.table’

The downloaded source packages are in
	‘/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/RtmpVQXmzz/downloaded_packages’
Warning message:
In install.packages(c("data.table", "jsonlite", "Matrix", "R6",  :
  installation of package ‘data.table’ had non-zero exit status
```

Full logs: https://github.com/microsoft/LightGBM/runs/1946047294?check_suite_focus=true.

--- a/.ci/test_r_package.sh
+++ b/.ci/test_r_package.sh
@@ -103,7 +103,7 @@ packages="c('data.table', 'jsonlite', 'Matrix', 'R6', 'testthat')"
 if [[ $OS_NAME == "macos" ]]; then
     packages+=", type = 'both'"
 fi
-Rscript --vanilla -e "options(install.packages.compile.from.source = 'both'); install.packages(${packages}, repos = '${CRAN_MIRROR}', lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'))" || exit -1
+Rscript --vanilla -e "options(install.packages.compile.from.source = 'never'); install.packages(${packages}, repos = '${CRAN_MIRROR}', lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'))" || exit -1
 
 cd ${BUILD_DIRECTORY}
 

--- a/.ci/test_r_package.sh
+++ b/.ci/test_r_package.sh
@@ -100,10 +100,12 @@ fi
 # Manually install Depends and Imports libraries + 'testthat'
 # to avoid a CI-time dependency on devtools (for devtools::install_deps())
 packages="c('data.table', 'jsonlite', 'Matrix', 'R6', 'testthat')"
+compile_from_source="both"
 if [[ $OS_NAME == "macos" ]]; then
     packages+=", type = 'binary'"
+    compile_from_source="never"
 fi
-Rscript --vanilla -e "options(install.packages.compile.from.source = 'never'); install.packages(${packages}, repos = '${CRAN_MIRROR}', lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'))" || exit -1
+Rscript --vanilla -e "options(install.packages.compile.from.source = '${compile_from_source}'); install.packages(${packages}, repos = '${CRAN_MIRROR}', lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'))" || exit -1
 
 cd ${BUILD_DIRECTORY}
 

--- a/.ci/test_r_package.sh
+++ b/.ci/test_r_package.sh
@@ -101,7 +101,7 @@ fi
 # to avoid a CI-time dependency on devtools (for devtools::install_deps())
 packages="c('data.table', 'jsonlite', 'Matrix', 'R6', 'testthat')"
 if [[ $OS_NAME == "macos" ]]; then
-    packages+=", type = 'both'"
+    packages+=", type = 'binary'"
 fi
 Rscript --vanilla -e "options(install.packages.compile.from.source = 'never'); install.packages(${packages}, repos = '${CRAN_MIRROR}', lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'))" || exit -1
 

